plugins/UsersManager/Controller.php

```diff
@@ -442,13 +443,10 @@ public function setIgnoreCookie()
         Piwik::checkUserHasSomeViewAccess();
         Piwik::checkUserIsNotAnonymous();
 
-        $salt = Common::getRequestVar('ignoreSalt', false, 'string');
-        if ($salt !== $this->getIgnoreCookieSalt()) {
-            throw new Exception("Not authorized");
-        }
+        Nonce::checkNonce(self::NONCE_SET_IGNORE_COOKIE);
 
         IgnoreCookie::setIgnoreCookie();
-        Piwik::redirectToModule('UsersManager', 'userSettings', array('token_auth' => false));
+        Piwik::redirectToModule('UsersManager', 'userSettings');
```

Comment:
```suggestion
        Piwik::redirectToModule('UsersManager', 'userSettings', ['nonce' => false]);
```

There should be no need to keep the nonce in the URL after the redirect. Should do no harm, but removing it makes all URLs a bit nicer.